File: 01/01.py
#!/usr/bin/env python


def accumulate(values, total=0):
    return (x + sum(values[:i]) for i, x in enumerate(values))

with open("input.txt", "r") as f:
    data = map(lambda x: 1 if x == "(" else -1, list(f.read()))

print("Part 1: {0}".format(sum(data)))
print("Part 2: {0}".format(list(accumulate(data)).index(-1) + 1))

File: 01/01.clj
(ns aoc-01)

(use 'clojure.java.io)

(defn to-numbers
  [x]
  (if (= \( x) 1
  (if (= \) x) -1)))

(def data
  (map to-numbers (seq (slurp "input.txt"))))

(defn part1
  []
  (reduce + data))

(defn part2
  []
  (loop [i 0 sum 0]
    (if (= sum -1)
      i
    (recur (inc i) (+ sum (nth data i))))))

(println (str "Part 1: " (part1) "\nPart 2: " (part2)))